sdks/python/apache_beam/utils/timestamp.py

```diff
@@ -151,7 +151,10 @@ def __int__(self):
   def __eq__(self, other):
     # Allow comparisons between Duration and Timestamp values.
     if not isinstance(other, Duration):
-      other = Timestamp.of(other)
+      try:
+        other = Timestamp.of(other)
+      except TypeError:
+        return NotImplemented
```

Comment:
Is  `assert_that(pcoll | 'StaticSides' >> beam.core.MapTuple(fn, 's1', 's2'),` from the test case evaluated differently without returning `NotImplemented` here?

Reply:
It's something I noticed during testing--it doesn't affect the passing test, but for comparing to a different type (e.g. None or a string) it does matter.